### ISSUE TITLE
[Bug]Last Year's Course Data Displayed for Upcoming Year

### ISSUE THREAD

**Issue by shfc** [compsci-adl/mytimetable#31]
### Description of the Bug

The timetable incorrectly shows last year's course data when upgrading.

### Reproduction Steps

Select a course for 2024. When the timetable is upgraded for 2025, the 2024 course data will be shown on the 2025 timetable.

### Expected Behaviour

Only display the courses scheduled for 2025 on the 2025 timetable.

### Additional Notes

<img width="1076" alt="image" src="https://github.com/user-attachments/assets/13e25d59-a28c-4ff9-8b72-66df329f2716">

**Comment by phoenixpereira** [comment]
@shfc We made some changes to the backend, and this should now be fixed. Can you please check on your end and verify this?

**Comment by shfc** [comment]
> @shfc We made some changes to the backend, and this should now be fixed. Can you please check on your end and verify this?

Yes, there is no display on the timetable for that course now. I guess the backend deletes the course data and makes it inaccessible. But I believe this is a frontend bug. I think the frontend should do some defensive programming and display the course depending on its property value rather than its accessibility from API. As it hardcodes the timetable for 2025, it shouldn't display other years' courses, no matter if they are accessible. Happy to discuss this further!

**Comment by shfc** [comment]
> Oh I see what you mean now, the issue is that the local storage had courses from 2024 that didn't get cleared when the year changed to 2025, so it still fetched and displayed them. To fix this, we would need to modify the code relating to local storage to also store the year for the timetable and clear the enrolled courses if the year changes.

Yes, this is what I mean. It is a good idea to clear the invalid courses.
